pkg/apitype/policy.go

```diff
@@ -57,6 +60,10 @@ type RequiredPolicy struct {
 
 	// Where the Policy Pack can be downloaded from.
 	PackLocation string `json:"packLocation,omitempty"`
+
+	// The configuration that is to be passed to the Policy Pack. This must be valid
+	// in accordance with the JSON schema for the Policy Pack's configuration.
+	Config map[string]interface{} `json:"config,omitempty"`
```

Comment:
What is the expected format for the keys here? Is it the unique policy rule? Or something like `<policy-rule>::<parameter-name>`? It might make sense to have the keys not be a general `string`, but something like `type PolicyRuleParameter string` or `type PolicyRuleParameter struct { ... }` (with the `struct` version supporting serialization to/from string).

That would hopefully allow for better type safety as we pass data around various places.

Reply:
I was thinking this would align with how AWSGuard is configured programmatically today:

```json
"config": {
    "all": "mandatory",
    "efs-encrypted": {
        "enforcementLevel": "advisory",
        "kmsKeyId": "the id"
    },
    "ec2-instance-no-public-ip": "advisory"
}
```

Each key would be the policy name and the value is either a string `enforcementLevel` or an object containing the config parameters for the policy (including an optional `enforcementLevel` parameter for specifying that).

I admit that allowing the value to be either a `string` or `object` adds some complexity, but we'll be accepting JSON config from `--policy-pack-config`, and it'd be nice if the structure of that matched the structure used programmatically, since folks will likely be hand-editing the JSON config in the file specified to `--policy-pack-config`, and having the service response match would just keep things simple as we can use the same implementation when parsing the config. (Though, right now in AWSGuard the policy names are [kebab-case](https://wiki.c2.com/?KebabCase), but we programmatically configure them using camelCase).

But if we don't like that, we could force the API response to always use an object for the value, which, for the above example, would look like:

```json
"config": {
    "all": {
        "enforcementLevel": "mandatory"
    },
    "efs-encrypted": {
        "enforcementLevel": "advisory",
        "kmsKeyId": "the id"
    },
    "ec2-instance-no-public-ip": {
        "enforcementLevel": "advisory"
    }
}
```

Reply:
Or be even more pedantic and force all properties to be their own object:

```json
"config": {
    "all": {
        "enforcementLevel": "mandatory"
    },
    "efs-encrypted": {
        "enforcementLevel": "advisory",
        "properties": {
            "kmsKeyId": "the id"
        }
    },
    "ec2-instance-no-public-ip": {
        "enforcementLevel": "advisory"
    }
}
```

Reply:
Oh man so many options.. my preference here would be the second approach, which i think is a balance of the two. Since people are likely writing this by hand we dont want to make it too tricky with nested objects.

```"config": {
    "all": {
        "enforcementLevel": "mandatory"
    },
    "efs-encrypted": {
        "enforcementLevel": "advisory",
        "kmsKeyId": "the id"
    },
    "ec2-instance-no-public-ip": {
        "enforcementLevel": "advisory"
    }
}
```